### PR TITLE
Code review and updates.

### PR DESCRIPTION
**Pull request by gjwgit**
Improve comments and noted all TODO items in order for anyone to complete the task.

Note this is merged back into dev, not main.

### CODE REVIEW

**Review by NinadBhat — CHANGES_REQUESTED** [review]
Are there issues corresponding to the TODOs? If yes, can you add the issue number to the TODO comments?

**Comment by gjwgit** [comment]
> Are there issues corresponding to the TODOs? If yes, can you add the issue number to the TODO comments?

Yes, the TODOs might be issues, but they tend to be small things that should be checked, comments updated, or a code fix to be done before a PR rather than necessarily requiring a full issue. Some do correspond to issues and will migrate.

I have also added a check for TODOs in the `make prep` just like the ignores and missing license files so they are less likely to be overlooked:

```
$ make todo
Files that include TODO items to be resolved:

lib/src/solid_login.dart:124:// TODO 20231228 gjw GET getIssuer FROM solid-auth BUT PERHAPS IT IS NOW PART OF
lib/src/solid_login.dart:156:      // TODO 20231228 gjw THE FOLLOWING FUNCTIONALITY FROM solid-auth NEEDS TO
lib/src/solid_login.dart:356:          // TODO 20231228 gjw ADD THE LINK AND THE VERSION ALL WITHIN THE SAME
lib/src/solid_login.dart:372:    // TODO 20231228 gjw SOMEONE PLEASE EXPLAIN THE RATIONALE BEHIND THE LOGIC
lib/src/solid_login.dart:398:      // TODO 20231228 gjw SOMEONE PLEASE EXPLAIN WHY USING A SafeArea HERE.
lib/src/constants/colours.dart:5:// TODO 20231228 gjw BETTER IF IT WAS BY FUNCTION RATHER THAN BY COLOUR!
make: [support/flutter.mk:191: todo] Error 1 (ignored)

------------------------------------------------------------------------
```

File: test/solid_test.dart [hunk 1]
```diff
@@ -1,11 +1,5 @@
...
 
-
 void main() {
-  test('adds one to input values', () {
-    final calculator = Calculator();
-    expect(calculator.addOne(2), 3);
-    expect(calculator.addOne(-7), -6);
-    expect(calculator.addOne(0), 1);
-  });
+  test('adds one to input values', () {});
```
Comment (NinadBhat) [inline test/solid_test.dart:4]: @gjwgit: Should this be removed?

**Review by NinadBhat — APPROVED** [review]
Looks good to me. Only one comment.

**Pull request merged**